src/Surfnet/StepupMiddleware/CommandHandlingBundle/Tests/EventHandling/BufferedEventBusTest.php

```diff
@@ -85,38 +85,44 @@ public function flushing_succesfully_empties_the_buffer_to_prevent_flushing_the_
     }
 
     /**
-     * This is tested by publishing an event when flushing and flushing again afterwards
-     *
      * @test
      * @group event-handling
      */
-    public function new_event_can_be_buffered_while_flushing()
+    public function an_event_caused_by_an_event_in_the_current_buffer_being_flushed_is_buffered_and_flushed_after_events_in_the_current_buffer()
     {
-        $event = $this->createDummyDomainMessage();
         $bus = new BufferedEventBus();
 
-        // in php7 replace this with anonymous class
-        $listener = new OnFirstCallPublishesToBusAndCountingCallsEventListener(
-            $bus,
-            new DomainEventStream([$event])
+        $firstEventInCurrentBuffer = $this->createDummyDomainMessage(null, 'First event in current buffer');
+        $eventCausedByFirstEvent = $this->createDummyDomainMessage(
+            null,
+            'Event caused by first event in current buffer'
         );
+        $secondEventInCurrentBuffer = $this->createDummyDomainMessage(null, 'Second event in current buffer');
 
-        $bus->subscribe($listener);
+        $expectedEventSequence = [$firstEventInCurrentBuffer, $secondEventInCurrentBuffer, $eventCausedByFirstEvent];
 
-        $bus->publish(new DomainEventStream([$event]));
+        $listener = new RecordEventsAndPublishToBusOnFirstCallEventListener(
+            $bus,
+            new DomainEventStream([$eventCausedByFirstEvent])
+        );
 
-        $this->assertEquals(0, $listener->callCount, 'Prior to the first flush, the callcount should be 0');
-        $bus->flush();
+        $bus->subscribe($listener);
+        $bus->publish(new DomainEventStream([$firstEventInCurrentBuffer, $secondEventInCurrentBuffer]));
         $bus->flush();
 
-        $this->assertEquals(2, $listener->callCount, 'After flushing twice, the callcount should be 2');
+        $actualEventSequence = $listener->getRecordedEvents();
+
+        $this->assertEquals($expectedEventSequence, $actualEventSequence);
     }
 
     /**
+     * @param string $id
+     * @param null $payload
      * @return DomainMessage
      */
-    private function createDummyDomainMessage()
+    private function createDummyDomainMessage($id = '1', $payload = null)
```

Comment:
`$id` has been introduced as param with default value, yet always receives null as value, so what is the goal?